pkg/controller/migration/convert/network.go

```diff
@@ -98,22 +121,157 @@ func handleCalicoCNI(c *components, install *Installation) error {
 	}
 
-	// CNI IPAM pools
-	if len(c.calicoCNIConfig.IPAM.IPv4Pools) != 0 {
-		return ErrIncompatibleCluster{"specifying ipv4_pools in cni config is not supported."}
+	if c.cniConfigName != "k8s-pod-network" {
+		return ErrIncompatibleCluster{fmt.Sprintf("%s, only 'k8s-pod-network' is supported as CNI name, found %s", errCtx, c.cniConfigName)}
 	}
 
 	// Other CNI features
 	if c.calicoCNIConfig.FeatureControl.FloatingIPs {
-		return ErrIncompatibleCluster{"floating IPs not supported"}
+		return ErrIncompatibleCluster{errCtx + ", floating IPs not supported"}
 	}
 	if c.calicoCNIConfig.FeatureControl.IPAddrsNoIpam {
-		return ErrIncompatibleCluster{"IpAddrsNoIpam not supported"}
+		return ErrIncompatibleCluster{errCtx + ", IpAddrsNoIpam not supported"}
+	}
+	if c.calicoCNIConfig.ContainerSettings.AllowIPForwarding {
+		return ErrIncompatibleCluster{errCtx + ", AllowIPForwarding not supported"}
 	}
 
 	return nil
 }
 
+func getNetworkingBackend(node CheckedDaemonSet, client client.Client) (string, error) {
+	netBackend, err := node.getEnv(ctx, client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
+	if err != nil {
+		return "", err
+	}
+
+	switch {
+	case netBackend == nil:
+		return "bird", nil
+	case *netBackend == "":
+		return "bird", nil
+	case *netBackend == "bird":
+		return "bird", nil
+	case *netBackend == "vxlan":
+		return "vxlan", nil
+	case *netBackend == "none":
+		return "none", nil
+	default:
+		return "", fmt.Errorf("CALICO_NETWORKING_BACKEND %s is not valid", *netBackend)
+	}
+}
+
+// subhandleCalicoIPAM checks all fields in the Calico IPAM configuration,
+// if any fields have unexpected values an error message will be returned.
+// The function tries to collect all the errors and report one message.
+// If there are no errors and the config can be added to the passed in 'install'
+// then nil is returned.
+func subhandleCalicoIPAM(netBackend string, cnicfg calicocni.NetConf, install *Installation) error {
+	switch netBackend {
+	case "bird":
+		install.Spec.CalicoNetwork.BGP = operatorv1.BGPOptionPtr(operatorv1.BGPEnabled)
+	case "vxlan":
+		install.Spec.CalicoNetwork.BGP = operatorv1.BGPOptionPtr(operatorv1.BGPDisabled)
+	default:
+		return ErrIncompatibleCluster{fmt.Sprintf("CALICO_NETWORKING_BACKEND %s is not valid", netBackend)}
+	}
+
+	// ignored fields:
+	//   - c.calicoCNIConfig.IPAM.Name
+
+	invalidFields := []string{}
+	if cnicfg.IPAM.Subnet != "" {
+		invalidFields = append(invalidFields, "ipam.subnet field is unsupported")
+	}
+
+	if len(cnicfg.IPAM.IPv4Pools) != 0 {
+		invalidFields = append(invalidFields, "ipam.ipv4pools field is unsupported")
+	}
+	if len(cnicfg.IPAM.IPv6Pools) != 0 {
+		invalidFields = append(invalidFields, "ipam.ipv6pools field is unsupported")
+	}
+
+	if len(invalidFields) > 0 {
+		return ErrIncompatibleCluster{"configuration could not be migrated: " + strings.Join(invalidFields, ",")}
+	}
+	return nil
+}
+
+const UnsupportedMsg = "is unsupported"
```

Comment:
Is this really worthwhile? Seems like it's used inconsistently throughout the file anyway (some spell it out, some use this constant).

IMO we should just remove the constant and spell out the words "is unsupported" - it's not that many extra characters!